### ISSUE TITLE
Change the tempo and pitch using ExoPlayer PlayBackParamters

### ISSUE THREAD

**Issue by Mouadabdelghafouraitali** [lincollincol/AudioTool#1]
Thank you so much for your awesome lib, I'm using `ExoPlayer `to change the audio `PlayBackParamters` : 

```
  public void setSpeed(float speed) {
        PlaybackParameters param = new PlaybackParameters(speed, (float) getCurrentPitch());
        player.setPlaybackParameters(param);
    }


    public void setPitch(float pitch) {
        PlaybackParameters param = new PlaybackParameters((float) getCurrentTempo(), pitch);
        player.setPlaybackParameters(param);
    }
```
Now I'm trying to save the modified audio into the storage using your library, but I'm facing an issue the output tempo and pitch  is completely different from the `exoplayer `:

**e.g.  PITCH : 0.86883157 | TEMPO : 0.8699582800000001**

In `exoplayer `those parameters should make the audio slower, when I use your lib as follows : 

```
                try {
                        AudioTool.getInstance(this)
                                .withAudio(copiedPath)
                                .changeAudioPitch(44100, pitch, tempo, null).saveCurrentTo(outputPath);

                    } catch (IOException e) {
                        e.printStackTrace();
                    }
```
The output playback audio becomes faster instead of slower, I will be very thankful if you help me out with this issue

**Comment by lincollincol** [comment]
Hello @Mouadabdelghafouraitali. Thank you for reporting this!

Import new library version v1.1 to use fixed functional.

In this version, there are three `changeAudioPitch()` methods:

1) `changeAudioPitch(sampleRate, pitchValue, pitch, onCompleteCallback)` - the most simple method.
pitchValue means your 0.86883157 value. pitch - one of enum `Pitch` type: `Pitch.UP` or `Pitch.DOWN` 

`Pitch.UP` means that your audio will be thinner(faster) and `Pitch.DOWN` means otherwise: rougher(slower) audio.

2) `changeAudioPitch(sampleRate, pitch, onCompleteCallback)` - the seconds one method without `Pitch` enum. Here should use **negative** or **positive** pitch value: -0.86883157 or 0.86883157.

Negative (-0.86883157) returns the same result as the method above as if it with parameter `Pitch.DOWN`
Positive (0.86883157) return the same result as the method above as if it with parameter `Pitch.UP`

3) `changeAudioPitch(sampleRate, deltaRate, deltaTempo,onCompleteCallback)` - the most flexible method. Here you can pass custom pitch and tempo values. Don't forget about the **negative** and **positive** pitch values. 

Negative or positive value for **deltaRate** produces a thinner or rougher audio effect.
Negative or positive value for **deltaTempo** produces a slower or faster audio effect.

**TL;DR**
Pitch.DOWN or (-pitchValue)- negative pitch value. Output audio will be rougher.
Pitch.UP or (pitchValue) - positive pitch value. Output audio will be thinner.

For your code it will look like this:
1) `changeAudioPitch(44100, 0.86883157f, Pitch.DOWN, null)`
2) `changeAudioPitch(44100, -0.86883157f, null)`
3) `changeAudioPitch(44100, -0.86883157f, -0.8699582800000001f, null)`

All these calls will produce the same result (output audio can be a little slower because tempo value != pitch value).

You can change pith in the interval from -12f to 12f.

**Comment by Mouadabdelghafouraitali** [comment]
@lincollincol  Thank you so much, I really appreciate your quick response, unfortunately, not working yet, the output is the same as the input audio :

```
 float pitch = (float) (getCurrentPitch());
 float tempo = (float) (getCurrentTempo());
 Log.d(TAG, getCurrentPitch() + " | " + getCurrentTempo()); //0.86883157 | TEMPO : 0.8699582800000001


  AudioTool.getInstance(this)
                            .withAudio(copiedPath)
                            .changeAudioPitch(44100, pitch, tempo, output -> {
                                mSubscriber.notifyData(outputPath);
                            });
```

When I run the code above, the output audio is the same as the input, nothing changed, I've tried to make the pitch and tempo values negative for test and the same thing happened.

**Comment by lincollincol** [comment]
@Mouadabdelghafouraitali  Did you update library version to 1.1?

**Comment by Mouadabdelghafouraitali** [comment]
@Mouadabdelghafouraitali Yes I did!

**Comment by lincollincol** [comment]
@Mouadabdelghafouraitali Try this one `changeAudioPitch(44100, 0.86883157f, Pitch.DOWN, null)` method to change audio pitch. Or try using a higher pitch value instead of 0.86883157f (2.86883157f e.g)

**Comment by Mouadabdelghafouraitali** [comment]
@lincollincol  Thank you, I know it's out of your lib usage, but in my app I use `ExoPlayer` to slow or fast the audio in real time, I use your lib to save the modified audio playback parameters (pitch speed) as a file, the issue is, I need to pass the `ExoPlayer` Pitch & Speed to your lib, so the output audio should be exactly the same as the `ExoPlayer`.

**Comment by lincollincol** [comment]
@Mouadabdelghafouraitali You mean that pitch functional in the AudioTool library works, but output file is not the same as ExoPlayer output?

**Comment by Mouadabdelghafouraitali** [comment]
In the version 1.0 of **AudioTool** the output file is not the same as `ExoPlayer `output, the output file of **AudioTool** becomes faster instead of slower, when I update to v1.1 the output of **AudioTool** is the same as the input (original song), nothing changed

**Comment by lincollincol** [comment]
@Mouadabdelghafouraitali Send me your audio file if you have the opportunity. I'd like to test it on my machine. Contacts:

Gmail: xlincollincolx@gmail.com
Telegram: @lincollincol
Or share the file with google disk and send me url.

**Comment by Mouadabdelghafouraitali** [comment]
@lincollincol  Massive thanks, I've sent the audio file with more details about the issue.

**Comment by lincollincol** [comment]
@Mouadabdelghafouraitali Maybe good news for you. This day, I was listening and comparing your audio produced with ExoPlayer and AudioTool library and decide that you should use `changeAudioSpeed()` method with **speed** value from ExoPlayer as xSpeed as a parameter.

```changeAudioSpeed(0.8699582800000001f, null)```

Your ExoPlayer methods `setSpeed()` and `setPitch()` only increase or decrease audio speed without pitch effects.

For example: 
if you pass as a parameter 0.1f or smaller to one of these methods - output audio will be very slow, but with **normal vocal voice**.
If you pass as a parameter -12 to AudioTool `changeAudioPitch()` method - output audio will be slower, but with **fat voice effect**.

Based on these outputs, I decide that ExoPlayer decrease audio speed and not change pitch. Then, you should use `changeAudioSpeed()` method from AudioTool library with the same speed values as a ExoPlayer `setSpeed()` or `setPitch()`

The second example for your values:

**SLOW** → PITCH : 0.91155733 | TEMPO : 0.91000528
**SLOWER** → PITCH : 0.86883157 | TEMPO : 0.8699582800000001
**SLOWEST** → PITCH : 0.82168468 | TEMPO : 0.82007197
**NIGHTCORE**  → PITCH : 1.1870272800000001 | TEMPO : 1.19996488

`changeAudioSpeed(0.91000528f, null) // SLOW`
`changeAudioSpeed(0.8699582800000001f, null) // SLOWER`
`changeAudioSpeed(0.82007197f, null) // SLOWEST`
`changeAudioSpeed(1.19996488f, null) // NIGHTCORE `

Hope it helps you. Waiting for your feedback!

**Comment by Mouadabdelghafouraitali** [comment]
@lincollincol  thank you so much for your help, I really appreciate it, that's all what I want for now, I will close this issue.
